xdp-redirect/Makefile

```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+
+XDP_TARGETS := xdp_redirect_basic.bpf xdp_redirect_cpumap.bpf xdp_redirect_devmap.bpf \
+	       xdp_redirect_devmap_multi.bpf
+
+TOOL_NAME := xdp-redirect
+
+USER_TARGETS := xdp-redirect
+
+LIB_DIR       = ../lib
+USER_LIBS     = -lm
+
+USER_EXTRA_C := xdp_redirect_basic.c xdp_redirect_cpumap.c xdp_redirect_devmap.c \
+		xdp_redirect_devmap_multi.c
+
+LIB_OBJS_NODEP := $(LIB_DIR)/util/xdp_sample.o
+
+LINKED_SKELS := xdp_redirect_basic.skel.h xdp_redirect_cpumap.skel.h xdp_redirect_devmap.skel.h \
+		xdp_redirect_devmap_multi.skel.h
+
+USER_GEN 	:= $(LINKED_SKELS)
+EXTRA_USER_DEPS := $(USER_GEN) $(USER_EXTRA_C)
```

Comment:
If you're teaching common.mk about `USER_EXTRA_C` anyway, this really shouldn't be needed...